core/jsAndWasmShared/src/LocalDateTime.kt

```diff
@@ -25,34 +27,36 @@ public actual class LocalDateTime internal constructor(internal val value: jtLoc
     public actual constructor(date: LocalDate, time: LocalTime) :
             this(jtLocalDateTime.of(date.value, time.value))
 
-    public actual val year: Int get() = value.year().toInt()
-    public actual val monthNumber: Int get() = value.monthValue().toInt()
+    public actual val year: Int get() = value.year()
+    public actual val monthNumber: Int get() = value.monthValue()
     public actual val month: Month get() = value.month().toMonth()
-    public actual val dayOfMonth: Int get() = value.dayOfMonth().toInt()
+    public actual val dayOfMonth: Int get() = value.dayOfMonth()
     public actual val dayOfWeek: DayOfWeek get() = value.dayOfWeek().toDayOfWeek()
-    public actual val dayOfYear: Int get() = value.dayOfYear().toInt()
+    public actual val dayOfYear: Int get() = value.dayOfYear()
 
-    public actual val hour: Int get() = value.hour().toInt()
-    public actual val minute: Int get() = value.minute().toInt()
-    public actual val second: Int get() = value.second().toInt()
+    public actual val hour: Int get() = value.hour()
+    public actual val minute: Int get() = value.minute()
+    public actual val second: Int get() = value.second()
     public actual val nanosecond: Int get() = value.nano().toInt()
 
     public actual val date: LocalDate get() = LocalDate(value.toLocalDate()) // cache?
 
     public actual val time: LocalTime get() = LocalTime(value.toLocalTime())
 
     override fun equals(other: Any?): Boolean =
-            (this === other) || (other is LocalDateTime && this.value == other.value)
+        (this === other) || (other is LocalDateTime && (this.value === other.value || this.value.equals(other.value)))
```

Comment:
There was some interoperability problem for equality of js objects in wasm. I put all special this code to the wrapper.

Reply:
The new version doesn't actually check `===` of JS objects, only of the wrappers, doesn't it? I don't notice any issues, though. Maybe it's alright not to add `===`? What kind of interoperability problem was that?

Reply:
> maybe it's alright not to add ===
Now it is `this === other || (other is XXX && value.equals(other.value)` in wrappers (almost every wrapper which has a equals in external declarations). You suggest to change it?

Reply:
I see that the original code is untouched; I just don't understand what purpose `this === other` has. If it's for JS interoperability, shouldn't it be `this.value === other.value` instead, since `value` is a JS object but `this` and `other` are not?

Reply:
But `this === other` is also has good decision because you dont need then casting for the `other`. We can do both.

Reply:
But all of this is purely as performance optimizations, right? Would simply doing `value.equals(other.value)` still work reliably?